File: app/src/test/java/com/mgaetan89/showsrage/fragment/AddShowFragment_GetSearchResultsTest.java
package com.mgaetan89.showsrage.fragment;

import com.google.gson.Gson;
import com.mgaetan89.showsrage.model.SearchResults;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class AddShowFragment_GetSearchResultsTest {
	@Parameterized.Parameter(0)
	public SearchResults searchResults;

	@Parameterized.Parameter(1)
	public int size;

	@Test
	public void getSearchResults() {
		assertThat(AddShowFragment.getSearchResults(this.searchResults).size()).isEqualTo(this.size);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		Gson gson = new Gson();

		return Arrays.asList(new Object[][]{
				{null, 0},
				{new SearchResults(), 0},
				{gson.fromJson("{data: null}", SearchResults.class), 0},
				{gson.fromJson("{data: {}}", SearchResults.class), 0},
				{gson.fromJson("{data: {results: null}}", SearchResults.class), 0},
				{gson.fromJson("{data: {results: []}}", SearchResults.class), 0},
				{gson.fromJson("{data: {results: [{}, {}]}}", SearchResults.class), 2},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/AddShowFragment_GetSearchResultsTest.kt
package com.mgaetan89.showsrage.fragment

import com.mgaetan89.showsrage.model.SearchResults
import com.mgaetan89.showsrage.network.SickRageApi
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class AddShowFragment_GetSearchResultsTest {
    @Parameterized.Parameter(0)
    var searchResults: SearchResults? = null

    @Parameterized.Parameter(1)
    var size: Int = 0

    @Test
    fun getSearchResults() {
        assertThat(AddShowFragment.getSearchResults(this.searchResults).size).isEqualTo(this.size)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            val gson = SickRageApi.gson

            return listOf(
                    arrayOf<Any?>(null, 0),
                    arrayOf<Any?>(SearchResults(), 0),
                    arrayOf<Any?>(gson.fromJson("{data: null}", SearchResults::class.java), 0),
                    arrayOf<Any?>(gson.fromJson("{data: {}}", SearchResults::class.java), 0),
                    arrayOf<Any?>(gson.fromJson("{data: {results: null}}", SearchResults::class.java), 0),
                    arrayOf<Any?>(gson.fromJson("{data: {results: []}}", SearchResults::class.java), 0),
                    arrayOf<Any?>(gson.fromJson("{data: {results: [{}, {}]}}", SearchResults::class.java), 2)
            )
        }
    }
}
